### PR TITLE
replace os with pathlib for ai_config.py

### PR DESCRIPTION
**Pull request by Back2Basics**
### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thouroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: scripts/ai_config.py [hunk 1]
```diff
@@ -37,7 +41,7 @@ def construct_full_prompt(self):
         # Construct full prompt
         full_prompt = f"You are {self.ai_name}, {self.ai_role}\n{prompt_start}\n\nGOALS:\n\n"
         for i, goal in enumerate(self.ai_goals):
-            full_prompt += f"{i+1}. {goal}\n"
```
Comment (nponeccop) [inline scripts/ai_config.py]: Drop unrelated formatting from this PR

File: scripts/ai_config.py [hunk 2]
```diff
@@ -1,6 +1,9 @@
+from pathlib import Path
+
 import yaml
+
```
Comment (nponeccop) [inline scripts/ai_config.py]: Don't add spaces or move imports around, the PR is not about it.

**Review by nponeccop — CHANGES_REQUESTED** [review]
Make the PR more focused:

- My pull request is atomic and focuses on a single change. - NOT TRUE
- I have not snuck in any "extra" small tweaks changes - NOT TRUE

File: scripts/ai_config.py [hunk 3]
```diff
@@ -37,7 +41,7 @@ def construct_full_prompt(self):
         # Construct full prompt
         full_prompt = f"You are {self.ai_name}, {self.ai_role}\n{prompt_start}\n\nGOALS:\n\n"
         for i, goal in enumerate(self.ai_goals):
-            full_prompt += f"{i+1}. {goal}\n"
+            full_prompt += f"{i + 1}. {goal}\n"
```
Comment (richbeales) [inline scripts/ai_config.py]: please avoid introducing needless spacing changes unrelated to your PR

**Comment by nponeccop** [comment]
We will be following the standard. You can submit the changes about improving standard compliance (which one? pep8? isort?) separately from this PR. Some of them are already elsewhere e.g. in #590

**Review by nponeccop — APPROVED** [review]

**Comment by nponeccop** [comment]
@Back2Basics There are conflicts again due to the merging of other PRs

**Comment by p-i-** [comment]
Hope this gets updated. Nice change.

**Pull request closed without merging**